pool.go

```diff
@@ -435,12 +432,7 @@ func (p *pool) fetchAndUpdate(ctx context.Context, node string, c cid.Cid, attem
 		}
 	}
 
-	var errTooManyRequests ErrSaturnTooManyRequests
-	if errors.As(err, &errTooManyRequests) {
-		err = &ErrSaturnTooManyRequests{
-			Node:       errTooManyRequests.Node,
-			RetryAfter: errTooManyRequests.RetryAfter,
-		}
```

Comment:
I have no idea why `err` was being replaced with an error of the same type with the same values. I just removed this.